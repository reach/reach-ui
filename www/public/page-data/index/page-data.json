{
  "componentChunkName": "component---src-pages-index-mdx",
  "path": "/",
  "webpackCompilationHash": "",
  "result": {
    "pageContext": { "isCreatedByStatefulCreatePages": true, "frontmatter": {} }
  }
}
