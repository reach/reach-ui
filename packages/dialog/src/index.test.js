import React from "react";
import renderer from "react-test-renderer";
import { Dialog } from "./index";

describe("rendering", () => {
  it("does not render children when not open", () => {
    let wrapper = renderer.create(
      <div lang="gr">
        <Dialog isOpen={false}>
          <div lang="en" />
        </Dialog>
      </div>
    );
    expect(wrapper.root.findAllByProps({ lang: "gr" })).toHaveLength(1);
    expect(wrapper.root.findAllByProps({ lang: "en" })).toHaveLength(0);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});

describe.skip("refs", () => {});

describe.skip("didMount", () => {});

describe.skip("willUnmount", () => {});
