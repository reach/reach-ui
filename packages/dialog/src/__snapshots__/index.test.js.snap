// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering does not render children when not open 1`] = `
<div
  lang="gr"
/>
`;
